server/src/main/java/io/druid/server/http/ClusterResource.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.server.http;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.inject.Inject;
+import com.sun.jersey.spi.container.ResourceFilters;
+import io.druid.discovery.DiscoveryDruidNode;
+import io.druid.discovery.DruidNodeDiscoveryProvider;
+import io.druid.guice.LazySingleton;
+import io.druid.server.http.security.StateResourceFilter;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.Set;
+
+/**
+ */
+@Path("/druid/coordinator/v1/cluster")
+@LazySingleton
+@ResourceFilters(StateResourceFilter.class)
+public class ClusterResource
+{
+  private final DruidNodeDiscoveryProvider druidNodeDiscoveryProvider;
+
+  @Inject
+  public ClusterResource(DruidNodeDiscoveryProvider discoveryProvider)
+  {
+    this.druidNodeDiscoveryProvider = discoveryProvider;
+  }
+
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response getClusterServers()
```

Comment:
It might be nice to have endpoints that just return one type of service and stuff do.  While maybe not necessary, it would be a relatively simple addition that benefits the operator (I think).  Or, take a parameter to filter which types you care about, or something similar.